### PR TITLE
Stream architecture rework p.1

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
(only UDP H264 is supported now)

### CODE REVIEW

**Review by patrickelectric — COMMENTED** [review]
Please be sure to remove all development and lost code around, I should review the PR later today.

**Comment by joaoantoniocardoso** [comment]
> Please be sure to remove all development and lost code around, I should review the PR later today.

Okay, I intentionally left some lost code in `src/stream/stream.rs` so you can better grasp to where its going to expand in the next parts. After your review I can execute the fixup.

File: src/stream/pipeline/fake_pipeline.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        // format for both its src the next element's sink pad.
+        // We are choosing "UYVY" because it is compatible by the
+        // application-rtp template capabilities.
+        // For more information: https://gstreamer.freedesktop.org/documentation/additional/design/mediatype-video-raw.html?gi-language=c#formats
+        let description = match &configuration.encode {
+            VideoEncodeType::H264 => {
+                format!(concat!(
+                        "videotestsrc pattern={pattern} is-live=true do-timestamp=true",
+                        " ! videoconvert",
+                        " ! video/x-raw,format=I420",
```
Comment (patrickelectric) [inline src/stream/pipeline/fake_pipeline.rs:57]: You can put width, height and framerate here, avoiding the next `video/x-h264`

**Review by patrickelectric — APPROVED** [review]

**Comment by patrickelectric** [comment]
We can continue as it's and get more improvements merged later

**Pull request merged**